packages/auth/src/auth.module.ts

```diff
@@ -1,85 +1,198 @@
 import { DynamicModule } from '@nestjs/common';
-import { getOptionsToken } from './auth.constants';
+import {
+  AUTH_STRATEGY_SERVICE_TOKEN,
+  getAuthorizationCodeFlowOptionsToken,
+  getTokenExchangeOptionsToken,
+} from './auth.constants';
 import {
   AccessMode,
+  AuthStrategy,
   ShopifyAuthModuleAsyncOptions,
   ShopifyAuthModuleOptions,
 } from './auth.interfaces';
 import { ShopifyAuthOfflineController } from './offline-auth/offline-auth.controller';
 import { ShopifyAuthOnlineController } from './online-auth/online-auth.controller';
 import { buildControllerHackForToken } from './utils/build-controller-hack-for-token.util';
 import { buildProvidersForToken } from './utils/build-provider-for-token.util';
+import { ShopifyTokenExchangeService } from './token-exchange/token-exchange.service';
+import { ShopifyAuthorizationCodeFlowAuthStrategyService } from './authorization-code-flow-auth-strategy.service';
+import { ShopifyTokenExchangeAuthStrategyOnlineService } from './token-exchange/token-exchange-auth-strategy-online.service';
+import { ShopifyTokenExchangeAuthStrategyOfflineService } from './token-exchange/token-exchange-auth-strategy-offline.service';
 
 export class ShopifyAuthModule {
   static forRootOnline(options: ShopifyAuthModuleOptions): DynamicModule {
+    if (options.authStrategy === 'TOKEN_EXCHANGE') {
+      return {
+        module: class ShopifyAuthTokenExchangeModule {},
+        global: true,
+        providers: [
+          ShopifyTokenExchangeService,
+          {
+            provide: AUTH_STRATEGY_SERVICE_TOKEN,
+            useClass: ShopifyTokenExchangeAuthStrategyOnlineService,
+          },
+          {
+            provide: getTokenExchangeOptionsToken(AccessMode.Online),
```

Comment:
we don't need both, one is enough. just thought it would be nice to explicitly distinguish since they represent different types. will go back to one token

Reply:
Just changed it to be a single token.